### PR TITLE
feat(hackathon-surplus): add surplus data to activity details

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

First part of hackathon surplus project.

Using backend's new api: https://barn.api.cow.fi/xdai/api/v1/users/0x4CBc3B6B16Ac4218169078aDd37947B76127E10B/total_surplus

Add surplus data to activity details 
image

Figma: https://www.figma.com/file/FrpBXkN4ZNuVhfXp4RJtI4/2023-Account-Modal?type=design&node-id=0-1&mode=design

# ⚠️ Notes ⚠️
- Styling not included
- API fn should be moved to SDK, will do so in an upcoming PR
- Right now only queries barn
- Tooltip missing
- Real link missing

# To Test

1. Connect wallet
2. Open activity details
* Surplus card should be displayed

### CODE REVIEW

File: src/utils/amountFormat/index.ts [hunk 1]
```diff
@@ -45,10 +45,10 @@ export function formatAmountWithPrecision(
   const decimalsSeparator = numberFormat.format(1.1)[1]
 
   // Trim the remainder up to precision
-  const reminderWithPrecission = remainder.toFixed(precision, undefined, Rounding.ROUND_HALF_UP)
+  const reminderWithPrecision = remainder.toFixed(precision, undefined, Rounding.ROUND_HALF_UP)
```
Comment (alfetopito) [inline src/utils/amountFormat/index.ts:48]: Unrelated, only typo fix

**Comment by elena-zh** [comment]
Hey @alfetopito , great!
For the future PR it would be also great to add a loading effect when numbers are loaded :)
1 question: should we hide the card when connected to an unsupported network? 
image

Thanks!

**Review by elena-zh — APPROVED** [review]
Approved!

File: package.json [hunk 2]
```diff
@@ -188,7 +188,7 @@
     "simple-sitemap-renderer": "^1.1.0",
     "styled-components": "^5.3.0",
     "styled-jsx": "5.1.2",
-    "swr": "^1.0.1",
+    "swr": "^2.2.0",
```
Comment (anxolin) [inline package.json:191]: 👀

File: src/api/gnosisProtocol/api.ts [hunk 3]
```diff
@@ -200,3 +200,27 @@ export async function getProfileData(chainId: ChainId, address: string): Promise
...
+export type TotalSurplusData = {
+  totalSurplus: string
+}
+
+// TODO: Move to the SDK
+export async function getSurplusData(chainId: ChainId, address: string): Promise<TotalSurplusData> {
+  console.log(`[api:${API_NAME}] Get surplus data for`, chainId, address)
+
+  const baseUrl = `https://barn.api.cow.fi/${NETWORK_TO_API_PREFIX[chainId]}/api`
+  const url = `/v1/users/${address}/total_surplus`
```
[the remark this answers is not in the document]
Comment (anxolin) [reply src/api/gnosisProtocol/api.ts:219]: well, i would hope we move the SDK here instead. I think eventually will be part of the monorepo

File: src/api/gnosisProtocol/hooks.ts [hunk 4]
```diff
@@ -117,3 +120,34 @@ export function useHasOrders(account?: string | null): boolean | undefined {
...
+
+export function useSurplusAmount(): UseSurplusAmountResult {
+  const { chainId, account } = useWalletInfo()
+  const nativeCurrency = useNativeCurrency()
+
+  const {
+    data: surplusAmount,
+    isLoading,
+    error,
+  } = useSWR<CurrencyAmount<Currency> | null>(['totalSurplus', chainId, account], async () => {
```
Comment (anxolin) [inline src/api/gnosisProtocol/hooks.ts:138]: what is the caching logic? 
Also, do you invalidate once the user has traded and got some?

File: src/api/gnosisProtocol/api.ts [hunk 5]
```diff
@@ -200,3 +200,27 @@ export async function getProfileData(chainId: ChainId, address: string): Promise
...
+
+export type TotalSurplusData = {
+  totalSurplus: string
+}
+
+// TODO: Move to the SDK
+export async function getSurplusData(chainId: ChainId, address: string): Promise<TotalSurplusData> {
+  console.log(`[api:${API_NAME}] Get surplus data for`, chainId, address)
+
+  const baseUrl = `https://barn.api.cow.fi/${NETWORK_TO_API_PREFIX[chainId]}/api`
```
Comment (shoom3301) [inline src/api/gnosisProtocol/api.ts:218]: Should we use `api.cow.fi` for prod?

Comment (alfetopito) [reply src/api/gnosisProtocol/api.ts:218]: Not currently available.
Will work on that once backend has it ready

File: src/api/gnosisProtocol/hooks.ts (hunk 4 above)
Comment (alfetopito) [reply src/api/gnosisProtocol/hooks.ts:138]: Good question, have not thought about that.
Will work on it in a follow up PR

Comment (anxolin) [reply src/api/gnosisProtocol/hooks.ts:138]: Thanks!

Comment (anxolin) [reply src/api/gnosisProtocol/hooks.ts:138]: linking https://github.com/cowprotocol/cowswap/pull/2781

**Pull request merged**